### PR TITLE
Anhnvvfx10105@funix.edu.vn: Add Junit Test for the class URLUtil.java

### PR DESCRIPTION
**Pull request by MonLumi**
Hi there, I'm studying about the JUnit test and I have an assignment need to contribute my JUnit test to any public repository. I found your repo is interesting and I add an JUnit test for the class URLUtil. Hope my pull request is approved and merge to your branch. 
Thanks.

### CODE REVIEW

**Comment by joelittlejohn** [comment]
Thanks @MonLumi. URLUtil is a good place to start if you're learning about unit testing. Since you're learning, I will add some comments on your code.

File: jsonschema2pojo-core/src/test/java/org/jsonschema2pojo/util/URLUtilTest.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    }
+
+    @Test
+    public void testParseUrlDefault() throws MalformedURLException {
+        String url = "file:this/is/url";
+        Assert.assertEquals(new URL(url), URLUtil.parseURL(url));
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testParseUrlException() {
```
Comment (joelittlejohn) [inline jsonschema2pojo-core/src/test/java/org/jsonschema2pojo/util/URLUtilTest.java:42]: I would rename this to `testParseThrowsExceptionWhenURLIsMalformed`.

File: jsonschema2pojo-core/src/test/java/org/jsonschema2pojo/util/URLUtilTest.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+    @Test
+    public void test2() {
+        expectedException.expect(IllegalArgumentException.class);
+        expectedException.expectMessage("Unable to parse source: " + "java:@$1242");
+        URLUtil.parseURL("java:@$1242");
+    }
+
+    @Test
+    public void testGetFileFromUrlCorrect() throws MalformedURLException {
```
Comment (joelittlejohn) [inline jsonschema2pojo-core/src/test/java/org/jsonschema2pojo/util/URLUtilTest.java:57]: No need to include the word 'Correct' here. Typically if you have a method `testGetFileFromUrl` that tests a method getFileFromURL, then it is expected that it tests the happy path.

Also, testGetFileFromUrl should be renamed to testGetFileFromURL (as this matches the method we are testing).

File: jsonschema2pojo-core/src/test/java/org/jsonschema2pojo/util/URLUtilTest.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        File file = new File(File.pathSeparatorChar + "testResource");
+        URI uri = file.toURI();
+        URL url = uri.toURL();
+        Assert.assertNotEquals(file, URLUtil.getFileFromURL(url));
+        Assert.assertEquals(file.getAbsolutePath(), URLUtil.getFileFromURL(url).getAbsolutePath());
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testGetFileFromUrlException() throws MalformedURLException {
```
Comment (joelittlejohn) [inline jsonschema2pojo-core/src/test/java/org/jsonschema2pojo/util/URLUtilTest.java:66]: Maybe rename to `testGetFileFromURLThrowsExceptionWhenURLIsMalformed`.

**Review by joelittlejohn — CHANGES_REQUESTED** [review]

File: jsonschema2pojo-core/src/test/java/org/jsonschema2pojo/util/URLUtilTest.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+    @Test
+    public void testParseProtocol() {
+        Assert.assertEquals(URLProtocol.HTTPS, URLUtil.parseProtocol("https://testhttps.protocol"));
+        Assert.assertEquals(URLProtocol.HTTP, URLUtil.parseProtocol("http://testhttp.protocol"));
+        Assert.assertEquals(URLProtocol.FILE, URLUtil.parseProtocol("file:/testfile/protocol"));
+        Assert.assertEquals(URLProtocol.CLASSPATH, URLUtil.parseProtocol("classpath:/test/classpath/class.java"));
+        Assert.assertEquals(URLProtocol.JAVA, URLUtil.parseProtocol("java:test/java"));
+        Assert.assertEquals(URLProtocol.RESOURCE, URLUtil.parseProtocol("resource:test/resource"));
+        Assert.assertEquals(URLProtocol.NO_PROTOCOL, URLUtil.parseProtocol("there are no protocol: at all"));
```
[the remark this answers is not in the document]
Comment (MonLumi) [reply jsonschema2pojo-core/src/test/java/org/jsonschema2pojo/util/URLUtilTest.java]: Thanks so much. This is the first time I knew that we can call directly the static method without prefixed by the class name

File: jsonschema2pojo-core/src/test/java/org/jsonschema2pojo/util/URLUtilTest.java (hunk 1 above)
Comment (MonLumi) [reply jsonschema2pojo-core/src/test/java/org/jsonschema2pojo/util/URLUtilTest.java]: Yeah, sorry I forgot to naming it before commit :D

File: jsonschema2pojo-core/src/test/java/org/jsonschema2pojo/util/URLUtilTest.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+
+    @Test
+    public void test2() {
+        expectedException.expect(IllegalArgumentException.class);
+        expectedException.expectMessage("Unable to parse source: " + "java:@$1242");
+        URLUtil.parseURL("java:@$1242");
+    }
```
[the remark this answers is not in the document]
Comment (MonLumi) [reply jsonschema2pojo-core/src/test/java/org/jsonschema2pojo/util/URLUtilTest.java]: I don't know much, but based on the built-in tool on IntelliJ, it suggest me should coverage these cases too

File: jsonschema2pojo-core/src/test/java/org/jsonschema2pojo/util/URLUtilTest.java (hunk 2 above)
Comment (MonLumi) [reply jsonschema2pojo-core/src/test/java/org/jsonschema2pojo/util/URLUtilTest.java]: Thanks, I renamed it

File: jsonschema2pojo-core/src/test/java/org/jsonschema2pojo/util/URLUtilTest.java (hunk 3 above)
Comment (MonLumi) [reply jsonschema2pojo-core/src/test/java/org/jsonschema2pojo/util/URLUtilTest.java]: Thanks, I renamed it

**Pull request closed without merging**